src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -315,7 +327,7 @@ def _to_dict(self):
         }
 
         if self._run_source == RunInfoSources.LOCAL:
-            result["flow_name"] = Path(str(self.flow)).resolve().name
+            result["flow_name"] = Path(str(self.flow)).resolve().name if not self._use_remote_flow else self._flow_name
```

Comment:
maybe make this a helper function? see it many times in the file

Reply:
Refined the logic, use `self._flow_name` instead.